src/msg/async/ProtocolV1.cc

```diff
@@ -1794,7 +1794,7 @@ CtPtr ProtocolV1::handle_client_banner(char *buffer, int r) {
   if (peer_addr.is_blank_ip()) {
     // peer apparently doesn't know what ip they have; figure it out for them.
     int port = peer_addr.get_port();
-    peer_addr.u = connection->socket_addr.u;
+    peer_addr.u = connection->target_addr.u;
```

Comment:
i think, we can drop the following `peer_addr.set_port(port);` now that we assign `connection->target_addr.u` to `peer_addr.u`.

and, it might be better if we can put
```c++
peer_addr.set_sockaddr(connection->get_peer_socket_addr().get_sockaddr());
```

[OT] can we change `AsyncConnection::get_peer_socket_addr()` so it returns an const reference of `target_addr`?
instead of accessing `u` member variable directly.

Reply:
the port is needed because the remote end might be a server bound to a specific port, but their outgoing connection is on a different port.